capture/scopes/waverunner/waverunner.py

```diff
@@ -397,8 +410,15 @@ def capture_and_transfer_waves(self):
         else:
             raise RuntimeError("WAVERUNNER: Error: Channel selection invalid")
         waves = self._parse_waveform(data)
-        # Put into uint8 range
-        waves = waves + 128
+        # Put into uint8/uint16 range.
+        if self.bit == 8:
+            waves = waves + 128
+        elif self.bit == 12:
+            # Scale from 16 to 12 bit. This is needed as the 12-bit oscilloscope
+            # returns a 16-bit value per sample.
+            waves = np.int16(waves / 65536 * 4096) + 2048
```

Comment:
From a performance viewpoint is there any difference between the division and multiplication and simple shift right and left again? I would assume that numpy handles this in a smart way but it might be worth checking.

Reply:
I did the experiment and both versions of doing this seem to be similar in terms of performance.